<<<<<<< HEAD
# `no-extra-non-null-assertion`

Disallow extra non-null assertion.
=======
# Disallow extra non-null assertion (`no-extra-non-null-assertion`)
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d

## Rule Details

Examples of code for this rule:

<!--tabs-->

### âŒ Incorrect

```ts
const foo: { bar: number } | null = null;
const bar = foo!!!.bar;
```

```ts
function foo(bar: number | undefined) {
  const bar: number = bar!!!;
}
```

```ts
function foo(bar?: { n: number }) {
  return bar!?.n;
}
```

### âœ… Correct

```ts
const foo: { bar: number } | null = null;
const bar = foo!.bar;
```

```ts
function foo(bar: number | undefined) {
  const bar: number = bar!;
}
```

```ts
function foo(bar?: { n: number }) {
  return bar?.n;
}
```

<<<<<<< HEAD
## Options

```jsonc
// .eslintrc.json
{
  "rules": {
    "@typescript-eslint/no-extra-non-null-assertion": "error"
  }
}
```

This rule is not configurable.

=======
## How to Use

```json
{
  "@typescript-eslint/no-extra-non-null-assertion": ["error"]
}
```

>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d
## Attributes

- [x] âœ… Recommended
- [x] ğŸ”§ Fixable
- [ ] ğŸ’­ Requires type information
