<<<<<<< HEAD
# `no-misused-new`

Enforce valid definition of `new` and `constructor`.
=======
# Enforce valid definition of `new` and `constructor` (`no-misused-new`)
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d

Warns on apparent attempts to define constructors for interfaces or `new` for classes.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### ❌ Incorrect

```ts
class C {
  new(): C;
}

interface I {
  new (): I;
  constructor(): void;
}
```

### ✅ Correct

```ts
class C {
  constructor() {}
}
interface I {
  new (): C;
}
```

## Options

<<<<<<< HEAD
```jsonc
// .eslintrc.json
{
  "rules": {
    "@typescript-eslint/no-misused-new": "error"
  }
}
```

This rule is not configurable.

=======
```json
{
  "@typescript-eslint/no-misused-new": "error"
}
```

>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d
## Related To

- TSLint: [no-misused-new](https://palantir.github.io/tslint/rules/no-misused-new/)

## Attributes

- [x] ✅ Recommended
- [ ] 🔧 Fixable
- [ ] 💭 Requires type information
