<<<<<<< HEAD
# `no-unused-vars`

Disallow unused variables.
=======
# Disallow unused variables (`no-unused-vars`)
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d

## Rule Details

This rule extends the base [`eslint/no-unused-vars`](https://eslint.org/docs/rules/no-unused-vars) rule.
It adds support for TypeScript features, such as types.

## How to Use

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "no-unused-vars": "off",
  "@typescript-eslint/no-unused-vars": ["error"]
}
```

## Options

See [`eslint/no-unused-vars` options](https://eslint.org/docs/rules/no-unused-vars#options).

<sup>

Taken with â¤ï¸ [from ESLint core](https://github.com/eslint/eslint/blob/main/docs/rules/no-unused-vars.md)

</sup>

## Attributes

- [x] âœ… Recommended
- [ ] ğŸ”§ Fixable
- [ ] ğŸ’­ Requires type information
