<<<<<<< HEAD
# `no-unused-expressions`

Disallow unused expressions.
=======
# Disallow unused expressions (`no-unused-expressions`)
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d

## Rule Details

This rule extends the base [`eslint/no-unused-expressions`](https://eslint.org/docs/rules/no-unused-expressions) rule.
It adds support for optional call expressions `x?.()`, and directive in module declarations.

## How to Use

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "no-unused-expressions": "off",
  "@typescript-eslint/no-unused-expressions": ["error"]
}
```

## Options

See [`eslint/no-unused-expressions` options](https://eslint.org/docs/rules/no-unused-expressions#options).

<sup>

Taken with ‚ù§Ô∏è [from ESLint core](https://github.com/eslint/eslint/blob/main/docs/rules/no-unused-expressions.md)

</sup>

## Attributes

- [ ] ‚úÖ Recommended
- [ ] üîß Fixable
- [ ] üí≠ Requires type information
