<<<<<<< HEAD
# `prefer-as-const`

Prefer usage of `as const` over literal type.
=======
# Prefer usage of `as const` over literal type (`prefer-as-const`)
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d

This rule recommends usage of `const` assertion when type primitive value is equal to type.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### ‚ùå Incorrect

```ts
let bar: 2 = 2;
let foo = <'bar'>'bar';
let foo = { bar: 'baz' as 'baz' };
```

### ‚úÖ Correct

```ts
let foo = 'bar';
let foo = 'bar' as const;
let foo: 'bar' = 'bar' as const;
let bar = 'bar' as string;
let foo = <string>'bar';
let foo = { bar: 'baz' };
```

<!--/tabs-->

<<<<<<< HEAD
## Options

```jsonc
// .eslintrc.json
{
  "rules": {
    "@typescript-eslint/prefer-as-const": "error"
  }
}
```

This rule is not configurable.

=======
>>>>>>> fdb7c92a283e1067c995a5190717b93d6197f83d
## When Not To Use It

If you are using TypeScript < 3.4

## Attributes

- [x] ‚úÖ Recommended
- [x] üîß Fixable
- [ ] üí≠ Requires type information
